torch/distributed/pipelining/_backward.py

```diff
@@ -201,6 +201,10 @@ def hook(grad_inputs):
                 inp.grad = dinputs[i]
             else:
                 inp.grad += dinputs[i]
+
+        for t in stage_outputs:
```

Comment:
i would love a more detailed comment here about why this helps and why it is safe to do at this point